docs: add config example with migration guide (#375)
graphprotocol/indexer-rs: docs/migration-config/README.md

# Migration guide

> In case you want to quickly migrate from the old stack into the new stack, you can use the following two configurations to help you to migrate without changing too much fields.

Reviewer: ```suggestion
If you want to migrate quickly from the old stack to the new stack, you can use the following two configurations to help you to do so without changing too many fields.
```

You just need to drop the config, use it in the `--config` args and add some additional environment variables.

> Please take a deeper look in the config, but 90% of the time you'll be safe just using it. If you find any issues using this config, feel free to open an issue.

Reviewer: If you check [a Markdown guide](https://google.github.io/styleguide/docguide/style.html#:~:text=Markdown%20content%20follows%20the%20residual,an%2080%2Dcharacter%20line%20limit.) they say .md is supposed be formatted so that if you used `cat` in a terminal and thus limited to 80 chars per line. Up to you.

## Missing configuration fields

The new stack requires you adding an escrow subgraph, similar to the current network subgraph field.

You can edit the config yourself and add the following fields:

```
[subgraphs.escrow]
query_url = "<escrow subgraph query url>"
deployment_id = "<escrow deployment id>"
```

You can also add these environment variables in your new setup:

```
INDEXER_SUBGRAPHS__ESCROW__QUERY_URL=<escrow subgraph query url>
INDEXER_SUBGRAPHS__ESCROW__DEPLOYMENT_ID=<escrow deployment id>
```

> We recommend you migrating overtime to be fully in the configuration and don't need to provide multiple environment variables that can stay in plaintext, this will make your work easier.

Reviewer: ```suggestion
We recommend gradually transitioning to a full configuration setup, so you won’t need to rely on multiple environment variables, especially those stored in plaintext. This will simplify your workflow and improve security.
```

## Booleans inside config

The configuration doesn't accept booleans as strings, so in case you had `INDEXER_SERVICE_SERVE_NETWORK_SUBGRAPH` environment variable, please update to `INDEXER_SERVICE__SERVE_NETWORK_SUBGRAPH` and use `true/false` as values (or you could update directly in the configuration).

> Please update the config accordingly. Also, check it out the explanation for each field in [config/minimal-config-example.toml](config/minimal-config-example.toml) and also [config/maximal-config-example.toml](config/maximal-config-example.toml)

Reviewer: ```suggestion
Please update the config accordingly. Also, check out the explanation for each field in [config/minimal-config-example.toml](config/minimal-config-example.toml) and also [config/maximal-config-example.toml](config/maximal-config-example.toml).
```
